buildfile.xml

```diff
@@ -200,12 +200,26 @@ Allow from 127.0.0.1
 			<available file="${project.root}/${htaccess.config}" />
 			<then><property name="config_prefix" value="${project.root}/" override="true" /></then>
 		</if>
-		<!-- Copy the htaccess -->
-		<copy tofile=".htaccess" file="${config_prefix}${htaccess.config}" overwrite="false">
-			<filterchain>
-				<expandproperties />
-			</filterchain>
-		</copy>
+		<!-- Copy the htaccess - checking for public directory layout -->
+		<if>
+			<available file="public" type="dir" />
+			<then>
+				<exec command="touch -c public/.htaccess" checkreturn="true" />
+				<copy tofile="public/.htaccess" file="${config_prefix}${htaccess.config}" overwrite="false">
+					<filterchain>
+						<expandproperties />
+					</filterchain>
+				</copy>
+				<copy tofile=".htaccess" file="${config_prefix}${htaccess-root-public.config}" overwrite="false" />
```

Comment:
Should this one be `overwrite="true"` since this will already exist if you're upgrading a project to use the public directory?

Reply:
Thanks Nath, but I think the false is the correct approach. Keeping in mind that false only applies if destination exists, and is newer than the source. (https://www.phing.info/guide/chunkhtml/CopyTask.html)


Just concerned that if it's set to true, it will override every time, not just on setup. Perhaps the upgrade scenario is best covered off by documentation and/or combined with other steps that the silverstripe-base project (or alternative) provides.

Reply:
Is there a reason to ever change the root `.htaccess`? I imagine any customisation would happen in the `public/.htaccess`, in which case `true` might make sense. Alternatively we can add an `exists public/.htaccess` so it only overwrites once. But yeah, it could be covered by documentation or an upgrade flag, we'll just need to make sure that exists and is easy to troubleshoot.